File: sdk/php/src/Api/KucoinWSService.php
<?php

namespace KuCoin\UniversalSDK\Api;

use KuCoin\UniversalSDK\Generate\Futures\FuturesPrivate\FuturesPrivateWs;
use KuCoin\UniversalSDK\Generate\Futures\FuturesPublic\FuturesPublicWs;
use KuCoin\UniversalSDK\Generate\Margin\MarginPrivate\MarginPrivateWs;
use KuCoin\UniversalSDK\Generate\Margin\MarginPublic\MarginPublicWs;
use KuCoin\UniversalSDK\Generate\Spot\SpotPrivate\SpotPrivateWs;
use KuCoin\UniversalSDK\Generate\Spot\SpotPublic\SpotPublicWs;
use React\EventLoop\LoopInterface;

/**
 * KucoinWSService provides WebSocket interfaces for Spot, Margin, and Futures trading.
 */
interface KucoinWSService
{
    /**
     * Starts the shared event loop to begin processing asynchronous WebSocket events.
     * Should be called once after all desired WebSocket clients are started.
     */
    public function startEventLoop();

    /**
     * Stops the shared event loop and halts all asynchronous event processing.
     */
    public function stopEventLoop();

    /**
     * Returns the shared ReactPHP event loop used by all WebSocket clients.
     *
     * @return LoopInterface
     */
    public function getLoop(): LoopInterface;

    /**
     * Returns the interface to interact with the Spot Trading WebSocket (public channel) API of KuCoin.
     *
     * @return SpotPublicWS
     */
    public function newSpotPublicWS(): SpotPublicWS;

    /**
     * Returns the interface to interact with the Spot Trading WebSocket (private channel) API of KuCoin.
     *
     * @return SpotPrivateWS
     */
    public function newSpotPrivateWS(): SpotPrivateWS;

    /**
     * Returns the interface to interact with the Margin Trading WebSocket (public channel) API of KuCoin.
     *
     * @return MarginPublicWS
     */
    public function newMarginPublicWS(): MarginPublicWS;

    /**
     * Returns the interface to interact with the Margin Trading WebSocket (private channel) API of KuCoin.
     *
     * @return MarginPrivateWS
     */
    public function newMarginPrivateWS(): MarginPrivateWS;

    /**
     * Returns the interface to interact with the Futures Trading WebSocket (public channel) API of KuCoin.
     *
     * @return FuturesPublicWS
     */
    public function newFuturesPublicWS(): FuturesPublicWS;

    /**
     * Returns the interface to interact with the Futures Trading WebSocket (private channel) API of KuCoin.
     *
     * @return FuturesPrivateWS
     */
    public function newFuturesPrivateWS(): FuturesPrivateWS;
}

File: sdk/java/src/main/java/com/kucoin/universal/sdk/api/KucoinWSService.java
package com.kucoin.universal.sdk.api;

import com.kucoin.universal.sdk.generate.futures.futuresprivate.FuturesPrivateWs;
import com.kucoin.universal.sdk.generate.futures.futurespublic.FuturesPublicWs;
import com.kucoin.universal.sdk.generate.margin.marginprivate.MarginPrivateWs;
import com.kucoin.universal.sdk.generate.margin.marginpublic.MarginPublicWs;
import com.kucoin.universal.sdk.generate.spot.spotprivate.SpotPrivateWs;
import com.kucoin.universal.sdk.generate.spot.spotpublic.SpotPublicWs;

/** KucoinWSService provides WebSocket interfaces for Spot, Margin, and Futures trading. */
public interface KucoinWSService {
  /**
   * Returns the interface to interact with the Spot Trading WebSocket (public channel) API of
   * KuCoin.
   *
   * @return SpotPublicWs
   */
  SpotPublicWs newSpotPublicWS();

  /**
   * Returns the interface to interact with the Spot Trading WebSocket (private channel) API of
   * KuCoin.
   *
   * @return SpotPrivateWs
   */
  SpotPrivateWs newSpotPrivateWS();

  /**
   * Returns the interface to interact with the Margin Trading WebSocket (public channel) API of
   * KuCoin.
   *
   * @return MarginPublicWs
   */
  MarginPublicWs newMarginPublicWS();

  /**
   * Returns the interface to interact with the Margin Trading WebSocket (private channel) API of
   * KuCoin.
   *
   * @return MarginPrivateWs
   */
  MarginPrivateWs newMarginPrivateWS();

  /**
   * Returns the interface to interact with the Futures Trading WebSocket (public channel) API of
   * KuCoin.
   *
   * @return FuturesPublicWs
   */
  FuturesPublicWs newFuturesPublicWS();

  /**
   * Returns the interface to interact with the Futures Trading WebSocket (private channel) API of
   * KuCoin.
   *
   * @return FuturesPrivateWs
   */
  FuturesPrivateWs newFuturesPrivateWS();
}
